### DISCUSSION TITLE
Hello - I would like to join the pyscaffold project

### DISCUSSION THREAD

**Discussion by julie777** [pyscaffold/pyscaffold#588] [Dev Discussions]
I just want to take a moment and introduce myself, as I would like to become a contributor to the project.

My name is Julie Jones. I have been a software developer for 30+ years and Python has become my passion. I've used it professionally for almost 20 years. I primarily worked at startup companies and really enjoyed helping them get past the first hump and implement repeatable processes and tools. I championed and instituted continuous integration at multiple companies, so I do have a significant background in the area of dev tooling.

I very much admire the initial goal of pyscaffold "make it simple for a new python developer to get up an running with a project that is set up for best practices development".

I've read the discussion about pyscaffold on the pypa discussion and like the idea, but I think there is/can be much more to it.

I settled on pyscaffold for my use and as a potential project to work on after reviewing more than a dozen Python tools that do packaging, dependency management, project management, virtualenv management and testing. I think that pyscaffold has the most potential for the future. It has an architecture that will support growing and following the pypa developments.

I hope the owner and maintainers with allow me to contribute.

Julie

**Comment by FlorianWilhelm** [comment]
Hi Julie, thanks for your kind words! 

We are always looking for new contributors and you are of course welcome. With your 30+ years of experience in software development, there is surely a lot you could help us with to improve PyScaffold. I guess the best way to start is by working on open issues, submitting PRs, or suggesting new ideas. Your support is highly appreciated.

Florian

**Comment by julie777** [comment]
@FlorianWilhelm @abravalheri 

It just occurred to me that although I introduced myself, and I have a linkedin profile, that I didn't give you one important detail.

### Why

Why do I want to work on this?
Why is this important to me?
Why am I starting so many discussions?

It has been on my personal todo list for more than 10 years to write a tool that makes python development easier. A tool that lets a novice get started without needing to know everything about the development tools and such. I want to make it easy for a person with an interest in learning python to just start coding.

At the same time I want the expert developer to be able to automate as much as possible, to get rid of the repetetive tasks. To do things as simple as creating a new python file in my working directory from a template so I start with everything where it needs to be and I don't get lazy and leave things out.  An example is writing a quick script to make some task easier. Do I start with an empty file and just start writing code? This means that as the file gets larger and the code gets more complex I slowly start adding a main function, then I add command line handling, etc. If creating a file with all the important stuff was just as easy as `touch myfile.py` I would never waste time "evolving" the structure of the code.

You guys already know about using the right tools, having all the precommit tasks wired up, simple test running...   I'm sure we all want that stuff. Push a button and the perfect project is ready to go.

When I read about pyscaffold on pypa I had an aha moment. I thought this is it. To be honest, when I first ran putup, I was a little disappointed. There was no venv. `putup --help` didn't tell me how to run tests. I quickly realized that there were no dependencies installed. A little more exploring and I found that there wasn't a way for me to say add "myfavorite tool" to every project I create. Or add my vscode config file to every project I create.

But I was very impressed by the documentation, the desire to be a pypa recommended tool, and after looking at the code I saw that the architecture had real potential. 

I already feel like part of the team. I don't know where the project goals will land relative to my "ideal", but I see a willingness to communicate and consider options and discuss things. _I love that!_

I have mentioned Hatch several times because I found it first. It looked on the surface like what I wanted to build. But, like black, it is very opinionated. The pypa discussion about pyscaffold made it apparent to me that the goal was to advance the python development ecosystem. Simple things like committing to pyproject.toml are important. (And not using a file with the name pyproject.toml, but in a completely non-compliant way like Poetry.) Being ready to replace setup tools with something better is important.

Python is fantastic. Python has great packages. What Python has never had is a good ecosystem for all of the mundane development lifecycle support and other languages are getting it right. I want to help Python stay at the top of the preferred languages list. 

Well, I drifted a little and started evangelizing. I like to work on things I feel passionate about. 😺

**Comment by julie777** [comment]
@abravalheri  @FlorianWilhelm 
Hi guys:

Please accept my apology. :anguished:

In my excitement to get started I made some mistakes.
1. I read the documentation like a typical user and not like a contributor who wants to learn everything possible.
2. I started creating discussions and asking questions, looking to you for information, when the answers could have been found in the doc.
3. I was spending most of my time reading source code and trying to learn everything I could from it (I guess based on the adage that source code it the one source of truth). But in this case the source code is only a partial truth because much of what PyScaffold is does not come from the code, but from the doc and from the tools it supports.

Thank you both for being so patient and supportive. I know that in many "organizations" the response would have been "RTFM".

As I said yesterday, before I continue participating I am going to read the entire doc site and learn everything that I can. Yesterday I got through to then end of the FAQ, taking copious notes, and reflecting on various aspects of the system. Hopefully, I will finish the rest today.

Regards,
Julie

**Comment by julie777** [comment]
I finished my review of the documentation. 
- I learned more than I would have imagined by treating it like something I will be tested on later.
- there were dozens of tiny corrections I would have made if it we as simple as clicking edit and deleting a letter or changing a word.
- There are several significant changes that I think we could make, some just organizational, that I believe would benefit our users. I will think about these for a while and then start a discussion.

I might have more after I process and organize my notes.

**Comment by abravalheri** [comment]
Thank you for all the effort @julie777! Really nice to see your interest on PyScaffold.

Regarding the comment: `there were dozens of tiny corrections I would have made if it we as simple as clicking edit and deleting a letter or changing a word.`, I just checked and the template we are using just added the "edit on Github" button 2 days ago (it is actually just a pencil icon on the top right, but it is there!)

I just re-run the build on Read the Docs, and checked it is there: https://pyscaffold.org/en/latest/#

Hopefully it will be useful!

**Reply by julie777** [reply]
I'll give it a try.
